lang/java/reef-common/src/main/java/org/apache/reef/driver/restart/DriverRestartManagerImpl.java

```diff
@@ -45,30 +45,39 @@
 
   private boolean restartBegan;
   private boolean restartCompleted;
+  private boolean isRestartSet;
```

Comment:
This set of mutable booleans smells like it might be better managed as a state machine. In any case, please document them as they all contain `restart` and are hard to distinguish.